pkg/webui/console/containers/device-onboarding-form/device-qr-scan-form-section/index.js

```diff
@@ -0,0 +1,191 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useCallback, useState } from 'react'
+import { defineMessages } from 'react-intl'
+import { useSelector, useDispatch } from 'react-redux'
+
+import QRModalButton from '@ttn-lw/components/qr-model-button'
+import { useFormContext } from '@ttn-lw/components/form'
+import Link from '@ttn-lw/components/link'
+import Icon from '@ttn-lw/components/icon'
+import ModalButton from '@ttn-lw/components/button/modal-button'
+
+import Message from '@ttn-lw/lib/components/message'
+
+import sharedMessages from '@ttn-lw/lib/shared-messages'
+
+import { readDeviceQr } from '@console/lib/device-claiming-parse-qr'
+
+import { listBrands } from '@console/store/actions/device-repository'
+
+import { selectDeviceBrands } from '@console/store/selectors/device-repository'
+import { selectSelectedApplicationId } from '@console/store/selectors/applications'
+
+const hexToDecimal = hex => parseInt(hex, 16)
+
+const m = defineMessages({
+  hasEndDeviceQR: 'Does your end device have a QR code? Scan it to speed registration.',
+  learnMore: 'Learn more',
+  scanEndDevice: 'Scan end device code',
+  deviceInfo: 'Device information',
+  resetQRCodeData: 'Reset QR code data',
+  resetConfirm:
+    'Are you sure you want to discard QR code data? The scanned device will not be registered and the form will be reset',
+  scanSuccess: 'QR code scanned successfully',
+})
+
+const qrDataInitialValues = {
```

Comment:
```suggestion
const qrDataInitialState = {
```
Initial values makes it sound like form data.